File: admin/src/app/modules/lesson/domain/interfaces/Lesson.ts
import {LessonStatusEnum} from "../enums/LessonStatusEnum";
import { UploadVideo } from 'app/modules/ud-upload/domain/interfaces/UploadVideo';

export interface Lesson{
   id?:number,
   name?:string,
   description?:string,
   duration?:string,
   courseId?:number,
   status?:string,
   createdDate?:Date,
   videos?: UploadVideo[]
}

File: api/DAL.Entities/Lesson.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DAL.Entities
{
    public class Lesson
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string Duration { get; set; }
        public int CourseId { get; set; }
        public DateTime CreatedDate { get; set; }
        public string Status { get; set; }

        public static class Statuses
        {
            public const string Active = "ACTIVE";
            public const string Deleted = "DELETED";

            public static string[] Available = new[]
            {
                Active,
                Deleted
            };
        }
    }
}
